docroot/profiles/custom/cgov_site/modules/custom/cgov_site_section/cgov_site_section.module

```diff
@@ -662,6 +661,10 @@ function entity_fields_modified(array $fields, EntityInterface $entity) {
   }
   if ($entity->original == null && $entity->status->getString() == "1" ) {
     return TRUE;
+  }
+   // Adding this condition for archived state when saving node from control moderation state.
```

Comment:
In this, $entity original status is returning different values from the node edit form save & node view moderation state. 
```
$entity->status->getString() !== $entity->original->status->getString()
node edit form (0 != '')
node view moderation state (0 != 0)
```
Because node view moderation state is failing, I have added the moderation state condition in the node entity.

Reply:
Ok, the diff context makes this harder to understand, but I see now.
Here is the issue, you should not have this line at all in this function. the function it is in is actually for testing if specific fields have been modified, per the name. (See below) So you should not return true in this case from this function, at all. You should figure out how to apply this check elsewhere.

```
/**
 * Checks to see if an entities values have changed compared to it's original.
 *
 * @param array $fields
 *   The fields which trigger a cache invalidation.
 * @param \Drupal\Core\Entity\EntityInterface $entity
 *   The updated entity.
 *
 * @return bool
 *   A TRUE or FALSE as to whether the fields were modified.
 */
function entity_fields_modified(array $fields, EntityInterface $entity) {
```

Reply:
Ok in discussing, this line is actually bad because a change for a node from editing reviewed to publish, without changing any field that affects the navigation display will cause the navigation to be invalidated. So this definitely needs to be removed.